model/host/host.go

```diff
@@ -3502,6 +3502,46 @@ func (h *Host) UnsetPersistentDNSInfo(ctx context.Context) error {
 	return nil
 }
 
+// SetSleepSchedule updates the host's sleep schedule. Note that if some fields
+// in the sleep schedule are not being modified, the sleep schedule must still
+// contain all the unmodified fields. For example, if this host is on a
+// temporary exemption when their daily schedule is updated, the new schedule
+// must still have the temporary exemption populated.
+func (h *Host) SetSleepSchedule(ctx context.Context, schedule SleepScheduleInfo) error {
```

Comment:
Is it true that the input struct here should omit `NextStopTime` and `NextStartTime` since they're being recalculated?

Reply:
It doesn't really matter if next stop/start are set or not in the input schedule since L3517-3518 will zero them out and then set them based on the updated schedule. They can be populated or not and it won't really matter.